src/Microsoft.Health.Fhir.Core/Features/Operations/Export/CreateExportRequestHandler.cs

```diff
@@ -31,18 +38,33 @@ public async Task<CreateExportResponse> Handle(CreateExportRequest request, Canc
         {
             EnsureArg.IsNotNull(request, nameof(request));
 
-            // TODO: Later we will add some logic here that will check whether a duplicate job already exists
-            // and handle it accordingly. For now we just assume all export jobs are unique and create a new one.
+            IReadOnlyCollection<KeyValuePair<string, string>> requestorClaims = _claimsExtractor.Extract();
 
-            var jobRecord = new ExportJobRecord(request.RequestUri);
+            // Compute the hash of the job.
+            var hashObject = new
+            {
+                request.RequestUri,
+                RequestorClaims = requestorClaims,
+                request.DestinationInfo,
+            };
 
-            // Store the destination secret
-            SecretWrapper result = await _secretStore.SetSecretAsync(jobRecord.SecretName, request.DestinationInfo.ToJson());
+            string hash = JsonConvert.SerializeObject(hashObject).ComputeHash();
```

Comment:
Should we enforce ordering on the claims collection so that it serializes the same every time?

Reply:
Good catch. Originally this was in JobRecord which had `requestorClaims?.OrderBy(claim => claim.Key, StringComparer.Ordinal).ToList();`. I moved the code out but forgot to move the ordering.